receiver/mongodbatlasreceiver/access_logs.go

```diff
@@ -418,3 +418,24 @@ func (alr *accessLogsReceiver) getClusterCheckpoint(groupID, clusterName string)
 	}
 	return nil
 }
+
+func (alr *accessLogsReceiver) setClusterCheckpoint(groupID string, clusterCheckpoint *accessLogStorageRecord) {
+	for key, value := range alr.record {
+		if key == groupID {
+			found := false
+			for idx, v := range value {
+				if v.ClusterName == clusterCheckpoint.ClusterName {
+					found = true
+					alr.record[groupID][idx] = clusterCheckpoint
+				}
+			}
+			if !found {
+				alr.record[groupID] = append(alr.record[groupID], clusterCheckpoint)
+			}
+			return
+		}
+	}
+
+	// If we get here, we didn't find the groupID in the map, so we need to add it
+	alr.record[groupID] = []*accessLogStorageRecord{clusterCheckpoint}
```

Comment:
We're iterating over a map just to check if a key exists. Can we do this instead?

```suggestion
	value, ok := alr.record[groupID]
	if !ok {
	alr.record[groupID] = []*accessLogStorageRecord{clusterCheckpoint}
	}
	
	if key == groupID {
		var found bool
		for idx, v := range value {
			if v.ClusterName == clusterCheckpoint.ClusterName {
				found = true
				alr.record[groupID][idx] = clusterCheckpoint
			}
		}
		if !found {
			alr.record[groupID] = append(alr.record[groupID], clusterCheckpoint)
		}
		return
	}
```